replica/table.cc

```diff
@@ -697,14 +697,22 @@ table::try_flush_memtable_to_sstable(lw_shared_ptr<memtable> old, sstable_write_
             co_return stop_iteration(_async_gate.is_closed());
         }
 
-        auto f = consumer(std::move(reader));
 
+        std::exception_ptr ex;
+        try {
+            co_await _compaction_manager.maybe_wait_on_current_compaction(this);
```

Comment:
this change doesn't belong to this patch.

I also need a few more words on the commit message explaining the motivation for the patch

Reply:
sorry for the noise, it's actually the last patch. github UI confuses me.

Reply:
It's not strictly prerequisite for blocking, but not running the flush in the correct scheduling group increases the chances of tripping the blocking behaviour. I see that I committed the wrong version of the code though - the function was moved to the table class and this commit on its own won't compile. Will address that shortly.

Reply:
we're already running flush in the correct scheduling group. see that actual flush is initiated here 'auto f = consumer(std::move(reader));', under memtable_scheduling_group, then its future is carried to post_flush.

Reply:
Ah, my bad, will roll that back. I misunderstood the code

Reply:
we can reduce the relevant code to, which shows that the actual flush inherits the scheduling group it was initiated with:

```
SEASTAR_THREAD_TEST_CASE(sched_group_test) {
    scheduling_group sg = create_scheduling_group("sg0", 100).get0();
    scheduling_group sg2 = create_scheduling_group("sg1", 100).get0();
    auto cleanup = defer([&] () noexcept { destroy_scheduling_group(sg).get(); });
    auto cleanup2 = defer([&] () noexcept { destroy_scheduling_group(sg2).get(); });

    auto try_flush = [&] () -> future<> {
        auto perform_flush = [&] () -> future<> {
            co_await sleep(1ms);
            BOOST_REQUIRE_EQUAL(internal::scheduling_group_index(current_scheduling_group()), internal::scheduling_group_index(sg));
        };

        BOOST_REQUIRE_EQUAL(internal::scheduling_group_index(current_scheduling_group()), internal::scheduling_group_index(sg));
        auto f = perform_flush();

        auto post_flush = [sg2, f = std::move(f)] () mutable -> future<> {
            BOOST_REQUIRE_EQUAL(internal::scheduling_group_index(current_scheduling_group()), internal::scheduling_group_index(sg2));
            co_await std::move(f);
        };

        co_await with_scheduling_group(sg2, std::ref(post_flush));
    };

    with_scheduling_group(sg, std::ref(try_flush)).get();
}
```